### PR TITLE
Rest-API-Advanced

### PR DESCRIPTION
**Pull request by Zastrozhyn**

### CODE REVIEW

File: Repository/src/main/java/com/epam/esm/dao/impl/TagDaoImpl.java [hunk 1]
```diff
@@ -2,115 +2,62 @@
...
     public Tag findTag(Long id) {
-        Tag tag = null;
-        try{
-            tag = jdbcTemplate.queryForObject(FIND_TAG_BY_ID_QUERY,
-                    new BeanPropertyRowMapper<>(Tag.class), id);
-        } catch (EmptyResultDataAccessException e) {
-            log.info(e.getMessage());
-        }
-        return tag;
+        return entityManager.find(Tag.class, id);
```
Comment (churaalena) [inline Repository/src/main/java/com/epam/esm/dao/impl/TagDaoImpl.java:41]: What about exception handling?

File: Repository/src/main/java/com/epam/esm/dao/impl/UserDaoImpl.java [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+import javax.persistence.PersistenceContext;
+import javax.persistence.criteria.CriteriaBuilder;
+import javax.persistence.criteria.CriteriaQuery;
+import javax.persistence.criteria.Root;
+import java.util.List;
+
+import static com.epam.esm.constant.StringConstant.ID;
+
+@Repository
+@Transactional
```
Comment (churaalena) [inline Repository/src/main/java/com/epam/esm/dao/impl/UserDaoImpl.java]: Why do we need this annotation?

Comment (Zastrozhyn) [reply Repository/src/main/java/com/epam/esm/dao/impl/UserDaoImpl.java]: Mistake. Already removed

File: Service/src/main/java/com/epam/esm/validator/UserValidator.java [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+import com.epam.esm.entity.User;
+import org.springframework.stereotype.Component;
+
+@Component
+public class UserValidator {
+    private static final int MAX_NAME_LENGTH = 30;
+    private static final int MIN_NAME_LENGTH = 2;
+    public boolean isValid(User user) {
+        String userName = user.getName();
+        if (userName != null) {
```
Comment (churaalena) [inline Service/src/main/java/com/epam/esm/validator/UserValidator.java:12]: Why not throwing exception?

File: Repository/src/main/java/com/epam/esm/dao/impl/UserDaoImpl.java (hunk 2 above)
Comment (Zastrozhyn) [reply Repository/src/main/java/com/epam/esm/dao/impl/UserDaoImpl.java]: Same mistake in the TagDaoImpl

File: Repository/src/main/java/com/epam/esm/dao/impl/TagDaoImpl.java (hunk 1 above)
Comment (Zastrozhyn) [reply Repository/src/main/java/com/epam/esm/dao/impl/TagDaoImpl.java:41]: Done

**Pull request merged**